napari/_vispy/overlays/scale_bar.py

```diff
@@ -173,7 +173,35 @@ def _on_box_change(self):
 
     def _on_text_change(self):
         """Update text information"""
-        self.node.text.font_size = self.overlay.font_size
+        # update the dpi scale factor to account for screen dpi
+        # because vispy scales pixel height of text by screen dpi
+        if self.node.text.transforms.dpi:
+            # use 96 as the reference dpi
+            dpi_scale_factor = 96 / self.node.text.transforms.dpi
```

Comment:
adjust for canvas pixel scale

```suggestion
            dpi_scale_factor = (
                (96 / self.node.text.transforms.dpi) *
                self.viewer.window._qt_viewer.canvas._scene_canvas.pixel_scale
            )
```

Reply:
I don't think this is correct? 
The pixel_scale should already be accounted for when the Visual is drawn to the canvas.
Notice that with this the text clips the box again, meaning that with this there are too many logical pixels in the text size.